build/source/driver/multi_driver.f90

```diff
@@ -521,42 +579,63 @@ real(dp),parameter        :: safety = 0.01_dp               ! safety factor to e
-   call vegPhenlgy(&
-                   ! input/output: data structures
-                   model_decisions,             & ! intent(in):    model decisions
-                   typeStruct%hru(iHRU),        & ! intent(in):    type of vegetation and soil
-                   attrStruct%hru(iHRU),        & ! intent(in):    spatial attributes
-                   mparStruct%hru(iHRU),        & ! intent(in):    model parameters
-                   progStruct%hru(iHRU),        & ! intent(in):    model prognostic variables for a local HRU
-                   diagStruct%hru(iHRU),        & ! intent(inout): model diagnostic variables for a local HRU
-                   ! output
-                   computeVegFlux(iHRU),        & ! intent(out): flag to indicate if we are computing fluxes over vegetation (.false. means veg is buried with snow)
-                   notUsed_canopyDepth,         & ! intent(out): NOT USED: canopy depth (m)
-                   notUsed_exposedVAI,          & ! intent(out): NOT USED: exposed vegetation area index (m2 m-2)
-                   err,message)                   ! intent(out): error control
-   call handle_err(err,message)
-   ! define the green vegetation fraction of the grid box (used to compute LAI)
-   diagStruct%hru(iHRU)%var(iLookDIAG%scalarGreenVegFraction)%dat(1) = greenVegFrac_monthly(timeStruct%var(iLookTIME%im))
-  end do  ! looping through HRUs
+ ! compute the exposed LAI and SAI and whether veg is buried by snow
+ if(istep==1)then  
+  do iGRU=1,nGRU
+   do iHRU=1,gru_struc(iGRU)%hruCount
+
+    ! get vegetation phenology
+    call vegPhenlgy(&
+                    ! input/output: data structures
+                    model_decisions,                & ! intent(in):    model decisions
+                    typeStruct%gru(iGRU)%hru(iHRU), & ! intent(in):    type of vegetation and soil
+                    attrStruct%gru(iGRU)%hru(iHRU), & ! intent(in):    spatial attributes
+                    mparStruct%gru(iGRU)%hru(iHRU), & ! intent(in):    model parameters
+                    progStruct%gru(iGRU)%hru(iHRU), & ! intent(in):    model prognostic variables for a local HRU
+                    diagStruct%gru(iGRU)%hru(iHRU), & ! intent(inout): model diagnostic variables for a local HRU
+                    ! output
+                    computeVegFluxFlag,             & ! intent(out): flag to indicate if we are computing fluxes over vegetation (.false. means veg is buried with snow)
+                    notUsed_canopyDepth,            & ! intent(out): NOT USED: canopy depth (m)
+                    notUsed_exposedVAI,             & ! intent(out): NOT USED: exposed vegetation area index (m2 m-2)
+                    err,message)                      ! intent(out): error control
+    call handle_err(err,message)
+
+    ! save the flag for computing the vegetation fluxes
+    if(computeVegFluxFlag)      computeVegFlux(iGRU)%hru(iHRU) = yes
+    if(.not.computeVegFluxFlag) computeVegFlux(iGRU)%hru(iHRU) = no
```

Comment:
Not possible (or easy). This was done in order to make use of the integer data structures. Do not currently have data structures for logical flags.